packages/next/client/page-loader.js

```diff
@@ -1,28 +1,45 @@
-/* global document */
+/* global document, window */
 import mitt from 'next-server/dist/lib/mitt'
 import unfetch from 'unfetch'
 
-// smaller version of https://gist.github.com/igrigorik/a02f2359f3bc50ca7a9c
-function supportsPreload (list) {
-  if (!list || !list.supports) {
-    return false
-  }
+function supportsPreload (el) {
   try {
-    return list.supports('preload')
-  } catch (e) {
+    return el.relList.supports('preload')
+  } catch {
     return false
   }
 }
 
-const hasPreload = supportsPreload(document.createElement('link').relList)
+const hasPreload = supportsPreload(document.createElement('link'))
+
+function createPreloadLink (url) {
+  const link = document.createElement('link')
+  link.rel = 'preload'
+  link.crossOrigin = process.crossOrigin
+  link.href = url
+  link.as = 'script'
+  document.head.appendChild(link)
+}
+
+// Retrieve a list of dependencies for a given route from the build manifest
+function getDependencies (route, _m) {
+  const isModern =
+    process.env.__NEXT_MODERN_BUILD &&
+    'noModule' in document.createElement('script')
+  if ((_m = window.__BUILD_MANIFEST) && (_m = _m[route])) {
+    return _m
+      .map(url => `/_next/${url}`)
+      .filter(url => /\.module\.js$/.test(url) === isModern)
```

Comment:
Now that the manifests are separate I think we can remove this.

Reply:
Done